Components/Payment/Provider/ActivePaymentMethodsProvider.php

```diff
@@ -0,0 +1,95 @@
+<?php
+
+namespace MollieShopware\Components\Payment\Provider;
+
+use Exception;
+use Mollie\Api\Exceptions\ApiException;
+use MollieShopware\Components\Config;
+use MollieShopware\Components\MollieApiFactory;
+use MollieShopware\Components\Payment\ActivePaymentMethodsProviderInterface;
+use Psr\Log\LoggerInterface;
+use Shopware\Models\Shop\Shop;
+
+class ActivePaymentMethodsProvider implements ActivePaymentMethodsProviderInterface
+{
+    /** @var Config */
+    private $config;
+
+    /** @var LoggerInterface */
+    private $logger;
+
+    public function __construct(Config $config, LoggerInterface $logger)
+    {
+        $this->config = $config;
+        $this->logger = $logger;
+    }
+
+    /**
+     * Returns a collection of active payment methods from the Mollie API.
+     *
+     * @param array $parameters
+     * @param Shop[]|null $shops
+     *
+     * @return array
+     */
+    public function getActivePaymentMethodsFromMollie($parameters = [], $shops = [])
+    {
+        $methods = [];
+
+        if (empty($shops)) {
+            return $methods;
+        }
+
+        foreach($shops as $shop) {
+            try {
+                $methods = $this->getActivePaymentMethodsForShop($shop, $methods, $parameters);
+            } catch (Exception $e) {
+                $this->logger->error(
+                    sprintf('Error when loading active payment methods from Mollie for shop %s', $shop->getName()),
+                    [
+                        'error' => $e->getMessage(),
+                    ]
+                );
+            }
+        }
+
+        return $methods;
+    }
+
+    /**
+     * @param Shop $shop
+     * @param array $methods
+     * @param array $parameters
+     * @return array
+     * @throws ApiException
+     */
+    private function getActivePaymentMethodsForShop(Shop $shop, array $methods = [], array $parameters = [])
+    {
+        $mollieApiFactory = new MollieApiFactory($this->config, $this->logger);
```

Comment:
can you please add the service factory from the DI to the constructor of this class?
reason is that we always have 1 single factory that can be used,
only in the installation a newable is required because its done before the DI stuff exists,...but that should already exist in such a way